Better document candidates post-processing

yugaego opened the pull request:
Closes #1502 .

@mandarm, please let us know if the adjusted documentation looks sufficiently clear to you with this PR; and/or if you have any other suggestions.

@dgutov, it is [already documented](https://github.com/company-mode/company-mode/blob/0ae7c293112248a0c36377d6859a95d216ae5e96/company.el#L436) that Company keeps "duplicates" with different annotations. Still, it seems that adding this clarification also to the `company-transformers` docstring may be a good choice.

Also note, that I set the manual's version to 1.0.3-snapshot. [Previously,](https://github.com/company-mode/company-mode/pull/1228#issuecomment-997294550) we considered an approach of publishing only releases. And I am still doubtful whether pre-release documentation should be published.  Still, I guess it could be useful to have this PR's changes published. But that's unless you're planning on releasing the next version soon, or we are not fine to have a snapshot feature (such as #1499) description published. (The latter consideration is one of the reasons I haven't got to updating Tips & Tricks with #1499 changes.)

Thanks.

dgutov commented on company.el:
```diff
@@ -537,9 +537,10 @@ even if the backend uses the asynchronous calling convention."
-Each function gets called with the return value of the previous one.
-The first one gets passed the list of candidates, already sorted and
-without duplicates."
+Each function is called with the return value of the previous one.  The
```
Could you reformat this by hand so that "The" goes to the next line?

yugaego replied:
Sure. Done.

dgutov commented on doc/company.texi:
```diff
@@ -1596,21 +1598,57 @@ system @uref{https://github.com/joaotavora/yasnippet, YASnippet}.
-listed below.
+@cindex :separate
+@cindex duplicates
+If a grouped backend contains the keyword @code{:separate}, use the
```
"use" -> "you can use" would be better, I guess.

dgutov commented on doc/company.texi:
```diff
@@ -1596,21 +1598,57 @@ system @uref{https://github.com/joaotavora/yasnippet, YASnippet}.
+@cindex backends
+@cindex grouped backends
+@cindex :separate
+The transformer functions are called sequentially, each with the
```
Using "in a sequence" instead of "sequentially" might make this a bit easier to read.

dgutov reviewed (changes requested):
Thank you! See some comments inline.

dgutov commented:
> And I am still doubtful whether pre-release documentation should be published. Still, I guess it could be useful to have this PR's changes published.

We could probably wait a couple of weeks and tag a new release: the changes since the last one are not too big. Little chance to break things anyway.

yugaego commented:
> > And I am still doubtful whether pre-release documentation should be published. Still, I guess it could be useful to have this PR's changes published.
> 
> We could probably wait a couple of weeks and tag a new release: the changes since the last one are not too big. Little chance to break things anyway.

Sounds good.

I've pushed the suggested changes.

dgutov commented:
Thanks!
